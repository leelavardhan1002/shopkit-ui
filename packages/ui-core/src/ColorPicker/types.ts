export interface Color {
  id: string;
  label: string;
  value: string;
  borderColor?: string;
  selectedBorderColor?: string;
  disabled?: boolean;
}

export interface CoreProps {
  colors: Array<Color>;
  colorValue?: Color;
  defaultColorValue?: Color;
  onColorChange?: (color?: Color) => void;
  disabled?: boolean;
}

export interface AppearanceProps {
  variant?: 'square' | 'pill';
  size?: 'small' | 'medium' | 'large';
  showTooltip?: boolean;
  showLabels?: boolean;
  borderRadius?: number;
  maxVisible?: number;
  maxVisibleLabel?: string;
}

export interface ColorPickerProps extends CoreProps, AppearanceProps {
  className?: string;
  'aria-label'?: string;
}
