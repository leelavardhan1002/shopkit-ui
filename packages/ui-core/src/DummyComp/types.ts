export interface DummyCompProps {
    // Add your props here
}