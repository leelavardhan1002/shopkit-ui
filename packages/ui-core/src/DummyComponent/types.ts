export interface DummyComponentProps {
    // Add your props here
}