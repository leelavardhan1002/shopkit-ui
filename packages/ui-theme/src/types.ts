import { coreTokens } from './data/coreThemeTokens';

declare module '@mui/material/styles' {
    interface Theme {
        sys: typeof coreTokens.sys;
    }
}

export interface ShopkitSysThemeOptions {
    palette?: typeof coreTokens.palette;
    sys: typeof coreTokens.sys;
}

export interface ShopkitThemeProviderProps {
    children: React.ReactNode;
    themeOverrides?: ShopkitSysThemeOptions;
}
