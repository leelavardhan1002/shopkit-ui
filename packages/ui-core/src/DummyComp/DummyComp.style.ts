import { styled } from '@mui/material/styles';

export const StyledDummyComp = styled('div')(({ theme }) => ({
    border: `1px solid ${theme.palette.divider}`,
    // Add your styles here
}));
