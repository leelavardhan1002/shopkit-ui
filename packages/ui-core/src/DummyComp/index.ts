export { DummyComp } from './DummyComp';
export type { DummyCompProps } from './types';