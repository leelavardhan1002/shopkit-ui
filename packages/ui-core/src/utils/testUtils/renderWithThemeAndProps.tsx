import { render, RenderOptions } from "@testing-library/react";
import { ShopKitThemeProvider } from "@shopkit/ui-theme";
import React from "react";

export const renderWithTheme = (component: React.ReactElement, options?: Omit<RenderOptions, 'wrapper'> & { wrapper?: React.ComponentType<{ children: React.ReactNode }> }) => {
    const Wrapper = ({ children }: { children: React.ReactNode }) => {
        const CustomWrapper = options?.wrapper;
        return (
            <ShopKitThemeProvider>
                {CustomWrapper ? <CustomWrapper>{children}</CustomWrapper> : children}
            </ShopKitThemeProvider>
        );
    };
    
    return render(component, { ...options, wrapper: Wrapper });
};
