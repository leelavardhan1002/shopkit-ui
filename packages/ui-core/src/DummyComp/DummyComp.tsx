import { DummyCompProps } from './types';
import { StyledDummyComp } from './DummyComp.style';

export const DummyComp: React.FC<DummyCompProps> = (props) => {
    return (
        <StyledDummyComp>
            DummyComp
        </StyledDummyComp>
    );
};