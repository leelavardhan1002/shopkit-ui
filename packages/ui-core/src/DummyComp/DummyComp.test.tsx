import { DummyComp } from './DummyComp';
import { renderWithTheme } from '../utils/testUtils/renderWithThemeAndProps';
import { DummyCompProps } from './types';

const defaultProps: Partial<DummyCompProps> = {};

const setupElement = (props: DummyCompProps) => {
    return renderWithTheme(<DummyComp {...props} />);
}

describe('DummyComp', () => {
    it('matches snapshot', () => {
        const { asFragment } = setupElement(defaultProps);
        expect(asFragment()).toMatchSnapshot();
    });
});
