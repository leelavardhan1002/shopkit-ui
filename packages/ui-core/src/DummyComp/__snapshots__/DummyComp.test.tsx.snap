// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DummyComp matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-xviiwl"
  >
    DummyComp
  </div>
</DocumentFragment>
`;
